x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -455,15 +451,17 @@ func kibanaClient(cfg kibanaConfig) (*kibana.Client, error) {
 }
 
 func findPolicy(cfg setupConfig, policies []kibanaPolicy) (*kibanaPolicy, error) {
+	policyID := ""
 	policyName := cfg.Fleet.TokenPolicyName
 	if cfg.FleetServer.Enable {
-		policyName = cfg.FleetServer.PolicyName
+		policyID = cfg.FleetServer.PolicyID
 	}
 	for _, policy := range policies {
-		if policy.Status != "active" {
-			continue
-		}
-		if policyName != "" {
+		if policyID != "" {
+			if policyID == policy.ID {
+				return &policy, nil
+			}
+		} else if policyName != "" {
```

Comment:
Is this still needed if we got rid of policy name?

Reply:
Yes because we still allow referencing the policy name for the TOKEN fetch, in the case that you are not using Elastic Agent not in Fleet Server mode.